addons/account/models/account_move_line.py

```diff
@@ -3232,6 +3232,59 @@ def action_automatic_entry(self, default_action=None):
         action['context'] = ctx
         return action
 
+    def action_add_from_catalog(self):
+        """ Will open the catalog view """
+        move = self.env['account.move'].browse(self.env.context.get('order_id'))
+        return move.action_add_from_catalog()
+
+    # -------------------------------------------------------------------------
+    # Catalog
+    # -------------------------------------------------------------------------
+    def _get_product_catalog_lines_data(self, **kwargs):
```

Comment:
For me, the code could be simplified. 
We can probably extends the behavior of `_is_readonly` in the right model to avoid arbitrary value in the elif.
The price and the quantity could be computed in the same way by refactoring a bit the code.

Reply:
The `len(self) == 1` is not really needed, we did that in Sales to avoid computing the price and sum the quantity.

Why did you remove the `self.product_id.ensure_one()' from your previous commit? In your use case, it's okay to sum the number of customizable desks and chairs? 

The catalog does not support editing products that are spread across multiple lines. We can't know on which line the customer wants to add or remove one item. The discount or the uom could also differ. In that case, in Sales, we show the price as if there was only 1 item in the order.